pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -363,7 +363,22 @@ func (c *Cluster) findInvalidMonitorPlacement(desiredMonCount int) (*NodeUsage,
 			}
 		}
 		if !monFoundInZone {
-			emptyZone = true
+			// emptyZone is used below to imply that an empty zone exists AND
+			// that zone contains a valid node (i.e. a node that new pods can be
+			// assigned). this check enforces that assumption by checking that
+			// the zone contains at least one valid node.
+			validNodeInZone := false
+			for _, nodeUsage := range nodeZones[zi] {
+				if nodeUsage.MonValid {
+					validNodeInZone = true
+					break
+				}
+			}
+			if validNodeInZone {
+				emptyZone = true
```

Comment:
Should this be named `emptyZoneWithValidNode`? Empty makes it sound like it doesn't have any nodes.

Reply:
the property of being empty (empty zone, empty node) refers to the presence or absence of monitor pods.